### PR TITLE
bug: fix regression in 4e6aacd

### LINKED ISSUE #382: Node app crashes for cluster_metrics API
Hi,
I am trying to implement cluster metrics using AggregatorRegistry. Earlier I tried it in my node application but after accessing the cluster_metric endpoint I used to get a blank response. So to understand how to implement it, I cloned the prom-client repository and tried running the cluster.js example which is included in the repo. But after accessing the cluster_metric on port number 3001, the app crashed with  **TypeError: metrics.forEach is not a function**. 

Screenshot 2020-07-31 at 11 29 30 AM

If anyone could please help me to find out what's going wrong.
Thanks.

### PR DESCRIPTION
**Pull request by zbjornson**
Fixes #382

This one's hard to test without forking processes...

### CODE REVIEW

File: lib/cluster.js [hunk 1]
```diff
@@ -178,12 +178,12 @@ function addListeners() {
 }
 
 // Respond to master's requests for worker's local metrics.
-process.on('message', message => {
+process.on('message', async message => {
```
Comment (SimenB) [inline lib/cluster.js:181]: Is it safe to make this async?

Comment (zbjornson) [reply lib/cluster.js]: I think so, but I wasn't handling the error. Just revised the commit and switched to a Promise.

File: lib/cluster.js [hunk 2]
```diff
@@ -180,11 +180,18 @@ function addListeners() {
...
+		Promise.all(registries.map(r => r.getMetricsAsJSON()))
+			.then(metrics => {
+				process.send({
+					type: GET_METRICS_RES,
+					requestId: message.requestId,
+					metrics,
+				});
+			})
+			.catch(err => {
+				console.error('Failed to get metrics', err);
```
Comment (SimenB) [inline lib/cluster.js:192]: should we use `process.send` to communicate _something_?

File: CHANGELOG.md [hunk 3]
```diff
@@ -36,6 +36,7 @@ project adheres to [Semantic Versioning](http://semver.org/).
 - fix: provide nodejs_version_info metric value after calling `registry.resetMetrics()` (#238)
 - fix: provide process_max_fds metric value after calling `registry.resetMetrics()`
 - fix: provide process_start_time_seconds metric value after calling `registry.resetMetrics()`
+- fix: fix regression in 4e6aacd (unreleased) (#382)
```
Comment (SimenB) [inline CHANGELOG.md:39]: do we need to add this? since it's not released I think it might just cause confusion for people using the changelog to migrate?

Comment (zbjornson) [reply CHANGELOG.md:39]: I agree. CI will fail without it but looks like we can merge despite that failure.

**Review by SimenB — APPROVED** [review]
I like it 👍

**Comment by siimon** [comment]
Awesome work! 👍

**Review by siimon — APPROVED** [review]

**Pull request merged**